examples/step-90/step-90.cc

```diff
@@ -0,0 +1,1262 @@
+    //  in the bulk cell quadrature points, which, upon normalization, give the
+    //  discrete normal vector in a bulk cell.
+    void assemble_cell_worker(
+      const typename DoFHandler<dim>::active_cell_iterator &cell,
+      ScratchData<dim>                                     &scratch_data,
+      CopyData<dim>                                        &copy_data)
+    {
+      const FEValues<dim> &fe_values = scratch_data.fe_values;
+      const FEValues<dim> &level_set_fe_values =
+        scratch_data.level_set_fe_values;
+
+      const std::vector<double> &cellJxW = fe_values.get_JxW_values();
+
+      std::vector<Tensor<1, dim>> grad_level_set(
+        level_set_fe_values.get_quadrature().size());
+      level_set_fe_values.get_function_gradients(level_set, grad_level_set);
+
+      const double factor =
+        stabilization_parameter *
+        std::pow(cell->minimum_vertex_distance(), stabilization_exponent);
+      for (const unsigned int q : fe_values.quadrature_point_indices())
+        {
+          const Tensor<1, dim> &normal =
+            (1.0 / grad_level_set[q].norm()) * grad_level_set[q];
+          for (const unsigned int i : fe_values.dof_indices())
+            for (const unsigned int j : fe_values.dof_indices())
+              copy_data.cell_matrix(i, j) +=
+                factor * (normal * fe_values.shape_grad(i, q)) *
+                (normal * fe_values.shape_grad(j, q)) * cellJxW[q];
+        }
+    }
+
+    void evaluate_cell_worker(
+      const typename DoFHandler<dim>::active_cell_iterator &cell,
+      ScratchData<dim>                                     &scratch_data,
+      CopyData<dim>                                        &copy_data)
+    {
+      double                     cell_stab_sqr = 0.0;
+      const FEValues<dim>       &fe_values     = scratch_data.fe_values;
+      const std::vector<double> &cellJxW       = fe_values.get_JxW_values();
+      const unsigned int n_q_points = fe_values.get_quadrature_points().size();
+      const FEValues<dim> &level_set_fe_values =
+        scratch_data.level_set_fe_values;
+
+      std::vector<Tensor<1, dim>> level_set_grad(n_q_points);
+      level_set_fe_values.get_function_gradients(level_set, level_set_grad);
+
+      std::vector<Tensor<1, dim>> sol_grad(n_q_points);
+      fe_values.get_function_gradients(solution, sol_grad);
+
+      const double factor =
+        stabilization_parameter *
+        std::pow(cell->minimum_vertex_distance(), stabilization_exponent);
+
+      for (const unsigned int q : fe_values.quadrature_point_indices())
+        {
+          const Tensor<1, dim> normal =
+            (1.0 / level_set_grad[q].norm()) * level_set_grad[q];
+
+          const double stabilization_at_point = normal * sol_grad[q];
+          cell_stab_sqr +=
+            factor * std::pow(stabilization_at_point, 2.0) * cellJxW[q];
+        }
+      copy_data.cell_stab_error_sqr = cell_stab_sqr;
```

Comment:
This is awkward: In one context you call the variable `cell_stab_error_sqr`, in the other you call it `cell_stab_sqr`. Could these be named the same way?